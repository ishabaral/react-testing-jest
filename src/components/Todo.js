import React from "react";

function Todo({ todo }) {
  const { id, title, completed } = todo;
  const work = <h1>{title}</h1>;
  const text = completed ? <strike>{work}</strike> : work;
  return <div data-testid={`todo-${id}`}>{text}</div>;
}

export default Todo;
