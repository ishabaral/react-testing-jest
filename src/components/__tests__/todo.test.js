import { cleanup, render, screen } from "@testing-library/react";
import Todo from "../Todo";
import * as renderer from "react-test-renderer";

afterEach(() => {
  cleanup();
});

test("should render non-completed todo ", () => {
  const todo = { id: 1, title: "wash dishes", completed: false };
  render(<Todo todo={todo} />);
  const todoElement = screen.getByTestId("todo-1");
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent("wash dishes");
  expect(todoElement).not.toContainHTML("<strike>");
});

test("should render completed todo ", () => {
  const todo = { id: 2, title: "wash car", completed: true };
  render(<Todo todo={todo} />);
  const todoElement = screen.getByTestId("todo-2");
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent("wash car");
  expect(todoElement).not.toContainHTML("<strike>");
});

test("matches snapshot 1", () => {
  const todo = { id: 1, title: "wash dishes", completed: false };
  const tree = renderer.create(<Todo todo={todo} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("matches snapshot 2", () => {
  const todo = { id: 2, title: "wash car", completed: true };
  const tree = renderer.create(<Todo todo={todo} />).toJSON();
  console.log(tree);
  expect(tree).toMatchSnapshot();
});
