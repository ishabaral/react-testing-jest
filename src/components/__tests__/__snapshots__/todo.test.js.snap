// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<div
  data-testid="todo-1"
>
  <h1>
    wash dishes
  </h1>
</div>
`;

exports[`matches snapshot 2 1`] = `
<div
  data-testid="todo-2"
>
  <strike>
    <h1>
      wash car
    </h1>
  </strike>
</div>
`;
